src/test/ui/const-generics/generic_const_exprs/unify-op-with-fn-call.rs

```diff
@@ -0,0 +1,24 @@
+// check-pass
+#![feature(generic_const_exprs, adt_const_params, const_trait_impl)]
+#![allow(incomplete_features)]
+
+#[derive(PartialEq, Eq)]
+struct Foo(u8);
+
+impl const std::ops::Add for Foo {
+    type Output = Self;
+
+    fn add(self, rhs: Self) -> Self::Output {
+        self
+    }
+}
+
+struct Evaluatable<const N: Foo>;
+
+fn foo<const N: Foo>(a: Evaluatable<{ N + N }>) {
+    bar::<{ std::ops::Add::add(N, N) }>();
+}
+
+fn bar<const N: Foo>() {}
```

Comment:
```suggestion
fn foo<const N: Foo>(a: Evaluatable<{ N + N }>) {
    bar::<{ std::ops::Add::add(N, N) }>();
}

fn bar<const N: Foo>() {}

fn baz<const N: usize>(a: [u8; N + N]) {
    biz::<{ std::ops::Add::add(N, N) }>();
}

fn biz<const N: usize>() {}
```
interested in seeing this tested for builtin types

Reply:
hmm yeah `Add::add(N, N)` doesnt unify with `N + N` for builtin types

Reply:
I think we could totally support this by just making `Node::BinOp(..)` for operator functions but probably ought to do that in a separate PR rather than alongside this mir -> thir refactor